DOC: rephrase the role of core devs (#28912)
scikit-learn/scikit-learn: doc/about.rst

The decision making process and governance structure of scikit-learn is laid
out in the :ref:`governance document <governance>`.

.. _authors:

The people behind scikit-learn
-------------------------------

Scikit-learn is a community project, developed by a large group of
people, all across the world. A few teams, listed below, have central
roles, however a complete list of contributors can be found `on github

> <https://github.com/scikit-learn/scikit-learn/graphs/contributors>`__.

Reviewer: A single underscore should be enough, no ?
```suggestion
<https://github.com/scikit-learn/scikit-learn/graphs/contributors>`_.
```

> <https://github.com/scikit-learn/scikit-learn/graphs/contributors>`__.

Author: I rephrased slightly to be more nuanced

> Core developer team

Reviewer: ```suggestion
Core Developer Team
```

....................

> The following people are currently core developers, in charge of

Reviewer: ok now I'm confused, I thought contributor was referring to all kind of contributions, hence the different teams which to me all are contributors. This section is only about the core devs. I'm happy with Maintainers Team as well (this is what I proposed in yesterday meeting but it was ruled out quickly).

> The following people are currently core developers, in charge of

Author: Yes, this is also what I had in mind: this subsection is not about comms, dev relations, etc...

> The following people are currently core developers, in charge of

Reviewer 2: The authors list includes maintainers only, so this line should be:

```suggestion
The following people are currently maintainers, in charge of
```

> The following people are currently core developers, in charge of

Author: You are right that it is called the "maintainers" group in the governance document

By the way, I find that the governance document is hard to read here because the section titles don't convey well the nesting:
https://scikit-learn.org/stable/governance.html#governance

Should I try to change, for instance using a listed description rather than section titles?

> The following people are currently core developers, in charge of

Reviewer: I agree that the nesting is not clear

> The following people are currently core developers, in charge of

Author: OK, I'm changing a bit the layout

The decision making process and governance structure of scikit-learn is laid
out in the :ref:`governance document <governance>`.

> .. _authors:

Reviewer 3: Should this tag be adapted? Shall we also credit the original academical authors of scikit-learn and its associated publication?

Author: We can adapt this tag if you think that it is important. I did not think it was, as it's purely an internal tag, and I was worried of doing my git grep wrong and thus missing some places pointing there.

I'm not in favor of crediting the original academic authors and the associated publication more than they already are: this list reflect the people involved in the first couple of years, which is a small fraction of the total road that got scikit-learn to where it is today.

Author: Actually, regarding the tags, I added it to make sure that old html links that point to the anchor "#authors" in html still worked.

I'm relearning an old lesson: each time one does a non obvious thing, it should come with a comment :)

The people behind scikit-learn
-------------------------------

Scikit-learn is a community project, developed by a large group of
people, all across the world. A few teams, listed below, have central
roles, however a more complete list of contributors can be found `on
github
<https://github.com/scikit-learn/scikit-learn/graphs/contributors>`__.

> Core Developer Team

Reviewer: To make this page better reflect the governance document I'd suggest
```suggestion
Maintainers Team
```

> Maintainers Team
> ....................

Reviewer: sorry, forgot that in my prev suggestion
```suggestion
Maintainers Team
................
```

The following people are currently maintainers, in charge of
consolidating scikit-learn's development and maintenance:

> .. include:: authors.rst

Reviewer 3: Do we also want to rename `authors.rst`?

Author: Done!

I hope that I did not screw something up here. I guess that we'll know next time we update the contributor list, eg when updating the emeritus
